sleepecg/classification.py

```diff
@@ -0,0 +1,756 @@
+        mapping string parameter names to values. Should not include
+        `records` and `n_jobs`.
+    mask_value : int, optional
+        Only required for keras models, as passed to `prepare_data_keras`
+        and `keras.layers.Masking`, by default `None`.
+    classifiers_dir : str | pathlib.Path, optional
+        Directory in which the `.zip` file is stored. If `None`
+        (default), the value is be taken from the configuration.
+
+    See Also
+    --------
+    load_classifier : Load classifiers saved with this function
+    """
+    if classifiers_dir is None:
+        classifiers_dir = get_config('classifiers_dir')
+
+    target_file = Path(classifiers_dir).expanduser() / name
+
+    model_type = model.__module__.split('.')[0]
+    classifier_info = {
+        'model_type': model_type,
+        'stages_mode': stages_mode,
+        'feature_extraction_params': feature_extraction_params,
+    }
+    if mask_value is not None:
+        classifier_info['mask_value'] = mask_value
+
+    with TemporaryDirectory() as tmpdir:
+        with open(f'{tmpdir}/info.yml', 'w') as infofile:
+            yaml.dump(classifier_info, infofile)
+
+        if model_type == 'keras':
+            model.save(f'{tmpdir}/classifier')
+        else:
+            raise ValueError(f'Saving model of type {type(model)} is not supported')
+
+        shutil.make_archive(target_file, 'zip', tmpdir)
+
+
+@dataclass
+class SleepClassifier:
+    """
+    Store a sleep classifier model and metadata.
+
+    Attributes
+    ----------
+    model : typing.Any
+        The classification model, should have `fit` and `predict` methods.
+    stages_mode : str
+        Identifier of the grouping mode. Can be any of `'wake-sleep'`,
+        `'wake-rem-nrem'`, `'wake-rem-light-n3'`, `'wake-rem-n1-n2-n3'`.
+    feature_extraction_params : dict[str, typing.Any]
+        The parameters passed to :func:`extract_features`, as a dictionary
+        mapping string parameter names to values. Does not include
+        `records` and `n_jobs`.
+    model_type : str
+        A string identifying the model type, e.g. `'keras'` or `'sklearn'`.
+        This is used by :func:`stage` to determine how to perform sleep
+        stage predictions.
+    mask_value : int, optional
+        Only required for models of type `'keras'`, as passed to
+        `prepare_data_keras` and `keras.layers.Masking`, by default `None`.
+    source_file : str, optional
+        The file from which the classifier was loaded using
+        :func:`load_classifier`, by default `None`.
+    """
+
+    model: Any
+    stages_mode: str
+    feature_extraction_params: Dict[str, Any]
+    model_type: str
+    mask_value: Optional[int] = None
+    source_file: Optional[str] = None
+
+    def __str__(self) -> str:
+        """
+        Create a string representation of a `SleepClassifier`.
+
+        Returns
+        -------
+        str
+            The string representation.
+        """
```

Comment:
I don't think we need a docstring for this standard method.

Reply:
Me neither, but the numpydoc setting for pydocstyle says otherwise 🤷 We could switch that off alternatively?

Reply:
How?

Reply:
By adding `add_ignore = D105` to the pydocstyle config in `setup.cfg`. `D105` refers to "Missing docstring in magic method" ([link](http://www.pydocstyle.org/en/stable/error_codes.html)).

Reply:
Seems worth it, WDYT? We won't ever need this I think.

Reply:
Likely not, I'll make the changes!